src/placeos-models/asset.cr

```diff
@@ -0,0 +1,64 @@
+require "./base/model"
+
+module PlaceOS::Model
+  class Asset < ModelBase
+    include RethinkORM::Timestamps
+
+    table :asset
+
+    attribute name : String, es_subfield: "keyword"
+    attribute category : String = ""
+    attribute description : String = ""
+
+    # # TO:DO field needs further defition
+    attribute size : String = ""
+
+    attribute purchase_date : Time
+    attribute good_until_date : Time?
+
+    attribute barcode : Int32
+    attribute brand : String = ""
+
+    # Saved as cents
+    attribute purchase_price : Int32
+
+    # Array of URLs to images for an asset
+    attribute images : Array(String) = [] of String
+
+    # URL of downloadable receipt
+    attribute invoice : String?
+
+    attribute quantity : Int32 = 0
+    attribute in_use : Int32 = 0
+
+    attribute other_data : JSON::Any = JSON::Any.new({} of String => JSON::Any), converter: JSON::Any::StringConverter, es_type: "object"
+
+    attribute consumable_assets : Array(Asset)? = [] of Asset
+
+    # Association
+    ###############################################################################################
+
+    has_many(
+      child_class: AssetInstance,
+      dependent: :destroy,
+      foreign_key: "asset_id",
+      collection_name: :asset_instances
+    )
+
+    has_many(
+      child_class: Asset,
+      foreign_key: "asset_id",
+      collection_name: :assets
+    )
+
+    # Validation
+    ###############################################################################################
+
+    validates :quantity, numericality: {minimum: 0}
```

Comment:
Will need to add this for `:in_use` as well

Reply:
`in_use` is validated to always be greater than `quantity` so that would cover it?

Reply:
Shouldn't `in_use <= quantity`?

Reply:
`in_use` could be negative